src/middleware/MiddlewareFactory.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @module MiddlewareFactory
+ */
+
+import { AuthenticationProvider } from "../IAuthenticationProvider";
+
+import { AuthenticationHandler } from "./AuthenticationHandler";
+import { HTTPMessageHandler } from "./HTTPMessageHandler";
+import { Middleware } from "./IMiddleware";
+import { RedirectHandlerOptions } from "./options/RedirectHandlerOptions";
+import { RetryHandlerOptions } from "./options/RetryHandlerOptions";
+import { RedirectHandler } from "./RedirectHandler";
+import { RetryHandler } from "./RetryHandler";
+import { TelemetryHandler } from "./TelemetryHandler";
+
+/**
+ * @private
+ * To check whether the environment is node or not
+ * @returns A boolean representing the environment is node or not
+ */
+const isNodeEnvironment = (): boolean => {
+	return new Function("try {return this === global;}catch(e){return false;}")(); // tslint:disable-line: function-constructor
```

Comment:
I'm curious as I don't know the answer. Why does this work better than checking if the node process is available or if there is a windows object?

Reply:
Yes. I think that should work